index.js

```diff
@@ -3,27 +3,49 @@ const download = require('download');
 const nodeNightlyVersion = require('node-nightly-version');
 const Configstore = require('configstore');
 const pkg = require('./package.json');
-const mv = require('fs').rename;
 const rm = require('rimraf');
 
+const extractDate = versionString => ~~versionString.split('nightly')[1].slice(0,8);
+const compVersion = (currentVersion, latestVersion) => extractDate(currentVersion) < extractDate(latestVersion);
+
 module.exports = {
-  install: () => {
-  		let osArchString;
-      return nodeNightlyVersion().then(latest => {
-      	const conf = new Configstore(pkg.name);
-        conf.set('version', latest);
-        const os = process.platform;
-        const arch = process.arch;
-        const type = 'nightly';
-        osArchString = `${latest}-${os}-${arch}`;
-        const url = `https://nodejs.org/download/${type}/${latest}/node-${osArchString}.tar`;
-        return download(url, __dirname, {extract:true});
-      })
+  install: (version) => {
+		let osArchString,nodeNightlyVer;
+		nodeNightlyVer = version !== undefined ? Promise.resolve(version) : nodeNightlyVersion();
+
+    return nodeNightlyVer.then(latest => {
+    	const conf = new Configstore(pkg.name);
+      conf.set('version', latest);
```

Comment:
Strange all these alignments looks ok in my sublime...
Though i will check for spacre vs tabs again.